packages/widget-v2/src/pages/SwapExecutionPage/useHandleTransactionTimeout.tsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { SwapExecutionState } from "./SwapExecutionPage";
+import { setOverallStatusAtom, swapExecutionStateAtom } from "@/state/swapExecutionPage";
+import { useAtomValue, useSetAtom } from "jotai";
+import { errorAtom, ErrorType } from "@/state/errorPage";
+
+export const useHandleTransactionTimeout = (swapExecutionState?: SwapExecutionState) => {
+  const { route, transactionDetailsArray, } = useAtomValue(
+    swapExecutionStateAtom
+  );
+  const setError = useSetAtom(errorAtom);
+  const setOverallStatus = useSetAtom(setOverallStatusAtom);
+  const [transactionTimeoutTimer, setTransactionTimeoutTimer] = useState<NodeJS.Timeout | undefined>();
+
+  useEffect(() => {
+    if (!route?.estimatedRouteDurationSeconds) return;
+    if (swapExecutionState === SwapExecutionState.pending && transactionTimeoutTimer === undefined) {
```

Comment:
add `&& route.txRequired === transactionDetailsArray.length`

Reply:
hmm

Reply:
if we want to change the logic so we wait until we get all of the transactionDetails before we start the timer, I think we can reduce the time to be like 1x estimatedRouteDuration

Reply:
yeah I agree, because it's better to make sure user have signed all tx

Reply:
yeah that makes sense